WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubfilterListItemMapper.kt

```diff
@@ -0,0 +1,94 @@
+package org.wordpress.android.ui.reader.subfilter
+
+import com.google.gson.Gson
+import org.wordpress.android.datasets.ReaderBlogTableWrapper
+import org.wordpress.android.models.ReaderBlog
+import org.wordpress.android.models.ReaderTagType
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.SITE
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.SITE_ALL
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.ItemType.TAG
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.Site
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.SiteAll
+import org.wordpress.android.ui.reader.subfilter.SubfilterListItem.Tag
+import org.wordpress.android.ui.reader.utils.ReaderUtilsWrapper
+import javax.inject.Inject
+
+private data class MappedItem(
+    val type: Int,
+    var feedId: Long = 0,
+    var blogId: Long = 0,
+    var tagSlug: String = "",
+    var tagType: Int = 0
+)
+
+class SubfilterListItemMapper @Inject constructor(
+    private val readerUtilsWrapper: ReaderUtilsWrapper,
+    private val readerBlogTableWrapper: ReaderBlogTableWrapper
+) {
+    fun fromJson(
+        json: String,
+        onClickAction: (filter: SubfilterListItem) -> Unit,
+        isSelected: Boolean
+    ): SubfilterListItem {
+        val mappedItem = if (json.isEmpty()) {
+            MappedItem(type = SITE_ALL.value)
```

Comment:
This is just my personal opinion, but I try to avoid using generic names for non-generic elements. A more specific name, like `MappedSubfilterListItem`  makes it easier to distinguish it from other elements and tells you what is it about.

Reply:
Agree 👍! Changed here f4da91bb363a762553322e5999f6b3863ca0f0c6